SwiftGit2/Repository.swift

```diff
@@ -558,4 +579,246 @@ final public class Repository {
 		let iterator = CommitIterator(repo: self, root: branch.oid.oid)
 		return iterator
 	}
+
+	// MARK: - Diffs
+
+	public func diff(for commit: Commit) -> Result<Diff, NSError> {
+		typealias Delta = Diff.Delta
+
+		guard !commit.parents.isEmpty else {
+			// Initial commit in a repository
+			return self.diff(from: nil, to: commit.oid)
+		}
+
+		var diffs = [OpaquePointer]()
+		var mergeDiff: OpaquePointer? = nil
+		for parent in commit.parents {
+			let diff: Result<OpaquePointer, NSError> = self.diff(from: parent.oid, to: commit.oid)
+			guard diff.error == nil else {
+				return Result<Diff, NSError>.failure(diff.error!)
+			}
+
+			diffs.append(diff.value!)
+
+			if mergeDiff == nil {
+				mergeDiff = diff.value!
+			} else {
+				let mergeResult = git_diff_merge(mergeDiff, diff.value)
+				guard mergeResult == GIT_OK.rawValue else {
```

Comment:
I think we could just do `git_object_free(diff)` here instead of keeping a list of `diffs`. That would reduce memory consumption and should also make the code a little clearer.

Reply:
nice catch :+1: